pywren/storage/storage.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import absolute_import
+import json
+
+from .storage_utils import *
+from .exceptions import *
+from .s3_service import S3Service
+
+
+class Storage(object):
+    """
+    A Storage object is used by executors and other components to access underlying storage service
+    without exposing the the implementation details.
+    Currently we only support S3 as the underlying service.
+    We plan to support other services in the future.
+    """
+
+    def __init__(self, config):
+        self.storage_config = config
+        self.prefix = config['storage_prefix']
+        self.service = config['storage_service']
+        if config['storage_service'] == 's3':
+            self.service_handler = S3Service(config['service_config'])
```

Comment:
can we call this storage backend ? might be better than service_handler. Also it might be better to have an explicit interface for storage backend - that makes it easier to code / test ?